### ISSUE TITLE
PinePhone Graphics support

### ISSUE THREAD

**Issue by time-killer-games** [enigma-dev/enigma-dev#2136]
ENIGMA supports all the specifications for getting its games running on the pine64.org manjaro / whatever edition pinephone, except the graphics we need are:

- OpenGL version string: 2.1 Mesa 20.1.5
- OpenGL shading language version string: 1.20
- OpenGL ES profile version string: OpenGL ES 2.0 Mesa 20.1.5
- OpenGL ES profile shading language version string: 
- OpenGL ES GLSL ES 1.0.16

Are there any plans to support this? Running games built with our current OpenGL ES 2 graphics system throws errors in terminal / zeniy / kdialog saying apparently we are using OpenGL ES 3 dependent code in our OpenGL ES 2 system and this really shouldn't be the case not just for PinePhone but also for better Android support.

**Comment by time-killer-games** [comment]
ENIGMA games look as they should on the PineBook Pro when I switch to use OpenGL 1.1 but the PinePhone as well as most other arm64 architecture devices do not support anything but some common version of OpenGL ES (v2+).

**Comment by fundies** [comment]
We haven't removed all the gles3 functions from gles 2 yet. It shouldn't take a ton of work but I'm not super fluent in opengl so I can't be  bothered

**Comment by time-killer-games** [comment]
Thanks a ton! I'll give that a try.

As a side note, if there is already an issue for OpenGL 3 code used in 2, this ticket can be closed.

**Comment by fundies** [comment]
Do you want to try this again now that android stuffs been merged?

**Comment by time-killer-games** [comment]
> Do you want to try this again now that android stuffs been merged?

Yeah that's awfully nice of you to suggest this, does anyone get on ENIGMA's irc channel anymore? Maybe you and/or whoever else would be willing to help could chat with me on there?

**Comment by time-killer-games** [comment]
> Do you want to try this again now that android stuffs been merged?

Also, on a slightly different subject, I'm going to unblock you if you'd rather talk on discord, I just wasn't sure if you wanted me to contunue contacting you on there, but I have a game that I programmed for a client in GM that I got working in ENIGMA (with the help of my video player extension from my fork), and its a really really good game. If I get my client's permission, since he is the copyright holder, would you, Robert, and Josh like private access to the source code + binary for occasional regression tests to do against the game manually? Unfortuneately, he doesn't want the game to be published in any way publicly, whether in source or binary form, so it can't be a CI test, but it is a nice game to test regressions for when done hands on, since you can't do it with CI for those reasons.

imo, the more games we can do this with, the better. and since it requires my libwebm extension, I'm happy to open a pr to get that into master, so it's not just in my fork, and you guys can actually make use of that project. This would be a great game to port to mobile anyway, since it was originally designed to run on mobile exclusively and is still available on google play.

https://drive.google.com/file/d/1davVngfwzSBaB8vSyA7Ht-SanglPVkea/view?usp=sharing

**Comment by time-killer-games** [comment]
Since the pinephone is linux anyway, it should be as simple as building with SDL2 and OpenGL ES 2 and transferring the binary to the phone while installing the deps on it, so it's more familiar than the android process, though that too has the wiki thanks to you and hugar.

**Comment by time-killer-games** [comment]
If I get it working on the pinephone, would it be worth mentioning enigma on the pine64 forum? I think a lot of the devs over there would like it,

**Comment by IntinteDAO** [comment]
Tomorrow I should get the Pinephone, I'll be able to test the Enigma and at the most send some information about it to a few places :P

**Comment by IntinteDAO** [comment]
@fundies @time-killer-games 

Sorry for waiting. The problem probably's still exists
https://0bin.net/paste/acwrT2K-#Xo3hXQJgnYDO8NNG46vm7GawAehQJpoNOJqWo8b+I0I

In theory compilation on Pinephone went "good", but binaries have 0 Bytes and Shader return a big log error

**Comment by fundies** [comment]
What version of gles shaders does the phone support?

**Comment by IntinteDAO** [comment]
OpenGL ES profile version string: OpenGL ES 2.0 Mesa 20.3.5
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16

**Comment by fundies** [comment]
Are you on discord. I have something to try?
